src/mmap_unix.rs

```diff
@@ -51,6 +51,49 @@ pub enum Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
+pub(crate) fn mmap_unix(
```

Comment:
No need to name this mmap_unix, you can name it mmap, and when you use it in the higher level abstractions you can either use it as mmap_unix::mmap, or do a `use mmap_unix::mmap as mmap_unix`. This way we use the namespace in which the function is defined. Same for `mmap_with_file`.

Why do you need these helpers? Can you explain it in the commit message as well? This is not immediately clear.